starter-code/src/StackDataStructure.js

```diff
@@ -1,2 +1,46 @@
-function StackDataStructure () {
-}
+class StackDataStructure {
+    constructor() {
+        this.stackControl = []
+        this.MAX_SIZE = 10
+    }
+
+    isEmpty() {
+        if (this.stackControl.length == 0) {
+            return true
+
+        } else {
+            return false
+        }
+
+    }
+
+    canPush() {
+        if (this.MAX_SIZE != this.stackControl.length) {
+            return true
+
+        } else {
+            return false
+        }
+
+    }
+
+    push(parameter) {
+        if (this.MAX_SIZE == this.stackControl.length) {
```

Comment:
La linea está bien, lo único que como arriba has hecho funciones que comparan estas cosas lo suyo es llamarlas para no hacer lo mismo todo el rato y solo llamar a la función.

la linea quedaría--->if (!this.canPush()) {